src/trap.c

```diff
@@ -3528,23 +3528,25 @@ drown()
 	if (Amphibious || (Swimming && !(u.divetimer == 0 && u.usubwater))) {
 		u.uinwater = 1;
 		if(uclockwork) u.uboiler = MAX_BOILER;
-		if (Swimming && !is_3dwater(u.ux-u.dx,u.uy-u.dy) && u.divetimer > 0 && !Is_waterlevel(&u.uz) && yn("Dive underwater?")=='y') {
+		if (Swimming && !is_3dwater(u.ux,u.uy) && u.divetimer > 0 && !Is_waterlevel(&u.uz) && yn("Dive underwater?")=='y') {
 			u.usubwater = 1;
-		} else if(is_3dwater(u.ux-u.dx,u.uy-u.dy)){
+		} else if(is_3dwater(u.ux,u.uy)){
 			u.usubwater = 1;
 		} else if(Amphibious && !Swimming){
 			u.usubwater = 1;
 		}
 		if (Amphibious &&  u.usubwater){
 			if (flags.verbose)
 				if(!Swimming) pline("But you aren't drowning.");
-			if (!Is_waterlevel(&u.uz)) {
+			if (!is_3dwater(u.ux, u.uy)) {
```

Comment:
This distinction was actually purposeful, the Plane of Water is an endless ocean with no top or bottom, while Dagon's level has you walking around on the seafloor.